src/mruby-widget-lib/src/gem.c

```diff
@@ -990,19 +1036,43 @@ mrb_remote_param_set_value(mrb_state *mrb, mrb_value self)
             next = (param->max-param->min)*value + param->min;
         br_set_value_int(param->br, param->uri, next);
     } else if(param->type == 'f') {
-        const float x = value;
         float out = 0;
-		if(param->scale && strstr(param->scale, "log")) {
-			if (param->min > 0) {
-				const float b = log(param->min);
-				const float a = log(param->max)-b;
-				out = expf(a*x+b);
-			} else { // min <= 0 ( e.g. envelope time param )
-				const float a = logf(1.0+4096.0f);
-				out = (expf(a*x)-1.0f)*param->max/4096.0f;
-			}
-		} else
+        if(param->scale && strstr(param->scale, "log")) {
+            // map [0,1] => [min,max]
+
+            // if logmin, clip values around [0,0.07]
+            if(param->logmin!=param->min && value > 0.f && value < 0.07f)
```

Comment:
no change required, though I typically write 0.0f < value && value < 0.07f as it's a bit closer to the ideal 0.0 < value < 0.07, which AFAIK only python does.